enclave-runtime/src/lib.rs

```diff
@@ -430,72 +431,67 @@ pub unsafe extern "C" fn produce_blocks(
 	blocks_to_sync_size: usize,
 	nonce: *const u32,
 ) -> sgx_status_t {
-	let mut blocks_to_sync_slice = slice::from_raw_parts(blocks_to_sync, blocks_to_sync_size);
-
-	let blocks_to_sync: Vec<SignedBlock> = match Decode::decode(&mut blocks_to_sync_slice) {
-		Ok(b) => b,
-		Err(e) => {
-			error!("Decoding signed blocks failed. Error: {:?}", e);
-			return sgx_status_t::SGX_ERROR_UNEXPECTED
-		},
+	let blocks_to_sync = match Vec::<SignedBlock>::decode_raw(blocks_to_sync, blocks_to_sync_size) {
+		Ok(blocks) => blocks,
+		Err(e) => return Error::Codec(e).into(),
```

Comment:
This is a new trait that I introduced in  `enclave_runtime::utils`. We should use this at other places too. It should also be moved to a separate crate because we can also use that in the `integritee_service`. However, I wanted to not introduce more changes currently.

Reply:
seems like a useful trait 🚀 - this can be used in many places in our code, great!